### PR TITLE
Damask improvements

### PR DESCRIPTION
**Pull request by MarDiehl**
two small improvements
- using new URL (damask.mpie.de instead of damask3.mpie.de)
- more specific handling of dependencies (closes #26680)

### CODE REVIEW

File: var/spack/repos/builtin/packages/damask-grid/package.py [hunk 1]
```diff
@@ -17,9 +17,10 @@ class DamaskGrid(CMakePackage):
 
     depends_on('pkgconfig', type='build')
     depends_on('cmake@3.10:', type='build')
+    depends_on('petsc+mpi+hdf5')
     depends_on('petsc@3.14.0:3.14,3.15.1:3.15', when='@3.0.0-alpha4')
     depends_on('petsc@3.14.0:3.14,3.15.1:3.16', when='@3.0.0-alpha5')
-    depends_on('hdf5+fortran')
+    depends_on('hdf5@1.12:+mpi+fortran')
```
Comment (sethrj) [inline var/spack/repos/builtin/packages/damask-grid/package.py]: Don't older versions of damask work with older versions of hdf5? If the new hdf5 version and hdf5 requirement are for a new version of damask you can leave those lines in place and add:
```python
    depends_on('petsc+mpi+hdf5', when='@xxxx:')
    depends_on('hdf5@1.12+mpi', when='@xxxx:')
```

Comment (MarDiehl) [reply var/spack/repos/builtin/packages/damask-grid/package.py]: Thanks for the hint. Even newer versions of DAMASK compile with HDF5 1.10, but we have ongoing issues with MPI-output which are partly resolved with 1.12. Parallel IO is therefore disabled for versions < 1.12. Using 1.10 should be considered as a fallback option.
From the discussion in the other merge request, I remember that version requirements should be hard requirements, not recommendations. After https://github.com/spack/spack/commit/dce4f161d4cebf95cd604cc1fbca284ea0e9c622, I therefore change the requirement to 1.10 (1.8 was never tested) and assume that most people will have a working default setup.

File: var/spack/repos/builtin/packages/damask-mesh/package.py [hunk 2]
```diff
@@ -17,22 +17,19 @@ class DamaskMesh(CMakePackage):
 
     depends_on('pkgconfig', type='build')
     depends_on('cmake@3.10:', type='build')
+    depends_on('petsc+mpi+hdf5')
     depends_on('petsc@3.14.0:3.14,3.15.1:3.15', when='@3.0.0-alpha4')
     depends_on('petsc@3.14.0:3.14,3.15.1:3.16', when='@3.0.0-alpha5')
-    depends_on('hdf5+fortran')
+    depends_on('hdf5@1.10:+mpi+fortran')
```
Comment (sethrj) [inline var/spack/repos/builtin/packages/damask-mesh/package.py:23]: The other aspect of this change is that now MPI is *required* (transitively) to build damask-mesh due to this and the new `petsc` requirement. Is that intended? Same with `damask-grid`. Do you mean to do something like:
```python
conflicts('hdf5+mpi', when='^hdf5@:1.10')
```
for the MPI error (previously your requirement for 1.12+), and
```python
depends_on('hdf5+mpi', when='^petsc +mpi')
```
to specify that it requires parallel HDF5 when built against parallel petsc?

Comment (MarDiehl) [reply var/spack/repos/builtin/packages/damask-mesh/package.py:23]: PETSc with MPI support is required for both solvers (mesh and grid), also for older versions of DAMASK. FFTW and HDF5 also need MPI support.

> Parallel IO is therefore disabled for versions < 1.12. Using 1.10 should be considered as a fallback option.

I have to correct myself: For older versions of HDF5 (<1.12), compression during parallel IO is disabled, not MPI-parallel IO in general. But disabling compression is a runtime operation. 

As far as I see, `petsc+mpi` requires `hdf5+mpi`,
https://github.com/spack/spack/blob/897616f073a4da58fd48213db44a0e2cdcd83f1c/var/spack/repos/builtin/packages/petsc/package.py#L98-L99

So `depends_on('hdf5@1.10:+mpi+fortran')` and `depends_on('hdf5@1.10:+fortran')` should result in the same setup. The first option is just more explicit.

**Review by sethrj — APPROVED** [review]

**Comment by sethrj** [comment]
Thanks for answering my questions and updating the package!

**Comment by MarDiehl** [comment]
> Thanks for answering my questions and updating the package!

thanks for maintaining spack!

**Pull request merged**